workflow/controller/operator.go

```diff
@@ -2437,7 +2437,7 @@ func (woc *wfOperationCtx) GetNodeTemplate(node *wfv1.NodeStatus) (*wfv1.Templat
 		}
 		return tmpl, nil
 	}
-	return woc.wf.GetTemplateByName(node.TemplateName), nil
+	return woc.wf.GetTemplateByName(wfutil.GetTemplateFromNode(*node)), nil
```

Comment:
the lines above this check for `node.TemplateRef`, so I think you might be able to simplify them. or the `GetTemplateFromNode` is redundant here

Reply:
I opted to undo this change.